zerver/lib/exceptions.py

```diff
@@ -801,3 +801,16 @@ class SlackImportInvalidFileError(Exception):
     def __init__(self, message: str) -> None:
         super().__init__(message)
         self.message = message
+
+
+class LockedError(JsonableError):
+    http_status_code: int = 409
+    data_fields = ["reason"]
+
+    def __init__(self, reason: str) -> None:
+        self.reason: str = reason
+
+    @staticmethod
+    @override
+    def msg_format() -> str:
+        return _("Conflict error: {reason}")
```

Comment:
Usually you don't want to do this sort of construction for internationalization; `reason` still also needs to be tagged for translation. Better would typically be to just have each `LockedError` supply a complete translated string, and have this class just provide the status code.

Reply:
Sounds correct